src/univers/versions.py

```diff
@@ -343,3 +343,173 @@ def __gt__(self, other):
         if not isinstance(other, self.__class__):
             return NotImplemented
         return gentoo.vercmp(self.value, other.value) > 0
+
+
+@attr.s(frozen=True, order=False, eq=False, hash=True)
+class LegacyOpensslVersion(Version):
+    """
+    Represent an Legacy Openssl Version .
+
+    For example::
+
+    # 1.0.1f|0.9.7d|1.0.2ac
+    """
+
+    @classmethod
+    def is_valid(cls, string):
+        return bool(cls.parse(string))
+
+    @classmethod
+    def parse(cls, string):
+
+        """
+        Returns the tuple containig the 4 segments (i.e major, minor, build, patch) of Legacy Version,
+        False if not valid Legacy Openssl Version.
+
+        For example::
+        >>> LegacyOpensslVersion.parse("1.0.1f")
+        (1, 0, 1, 'f')
+        >>> LegacyOpensslVersion.parse("1.0.2ac")
+        (1, 0, 2, 'ac')
+        >>> LegacyOpensslVersion.parse("2.0.2az")
+        False
+        """
+
+        # All legacy base version of openssl that ever exited/exists.
+        all_legacy_base = (
+            "0.9.1",
+            "0.9.2",
+            "0.9.3",
+            "0.9.4",
+            "0.9.5",
+            "0.9.6",
+            "0.9.7",
+            "0.9.8",
+            "1.0.0",
+            "1.0.1",
+            "1.0.2",
+            "1.1.0",
+            "1.1.1",
+        )
+        # check if starts with a valid base
+        if not string.startswith(all_legacy_base):
+            return False
+
+        segments = string.split(".")
+        if not len(segments) == 3:
+            return False
+        major, minor, build = segments
+        major = int(major)
+        minor = int(minor)
+        if build.isdigit():
+            build = int(build)
+            patch = ""
+        else:
+            patch = build[1:]
+            build = int(build[0])
+        if patch and patch[0].isdigit():
```

Comment:
```suggestion
             if patch and patch[0].isdigit():
```
this should be indented as this is never needed when `patch=""`

Reply:
Indeed, will replace with
```python
        else:
            patch = build[1:]
            build = int(build[0])
            if patch[0].isdigit():
```